_usage/query_planning.md

```diff
@@ -0,0 +1,180 @@
+FROM            dbo.Orders LEFT OUTER JOIN
+                         dbo.Employees ON dbo.Orders.EmployeeID = dbo.Employees.EmployeeID
+order by dbo.Orders.EmployeeID, dbo.Orders.CustomerID, dbo.Orders.OrderID
+
+SET BABELFISH_SHOWPLAN_ALL OFF
+GO
+```
+
+
+**Query Plan display options**
+
+You can use settings that are similar to the PostgreSQL [EXPLAIN and EXPLAIN ANALYZE statement settings](https://www.postgresql.org/docs/current/sql-explain.html) to control the type of information that is displayed with your query plan.  To query Babelfish for a list of settings and their current values, use the following command:
+
+```
+SELECT name,
+       setting,
+       short_desc,
+       vartype,
+       enumvals,
+       boot_val,
+       reset_val,
+       pending_restart
+FROM pg_settings
+WHERE name LIKE '%babelfishpg_tsql.explain%';
+```
+
+| name                               | setting | short_desc                          | vartype | enumvals | boot_val | reset_val | pending_restart |
+| ---------------------------------- | ------- | ----------------------------------- | ------- | -------- | -------- | --------- | --------------- |
+| babelfishpg_tsql.explain_buffers | off | Include information on buffer usage bool | off | off | FALSE |
+| babelfishpg_tsql.explain_costs | on | Include information on estimated startup and total cost | bool | on | on | FALSE |
+| babelfishpg_tsql.explain_format text | Specify the output format, which can be TEXT, XML, JSON, or YAML | enum | {text,xml,json,yaml} | text | text | FALSE |
+| babelfishpg_tsql.explain_settings | off | Include information on configuration parameters | bool | off | off | FALSE |
+| babelfishpg_tsql.explain_summary | off | Include summary information (e.g., totaled timing information) after the query plan | bool | off | off | FALSE |
+| babelfishpg_tsql.explain_timing | off | Include actual startup time and time spent in each node in the output | bool | off | off | FALSE |
+| babelfishpg_tsql.explain_verbose | off | Display additional information regarding the plan | bool | off | off | FALSE |
+| babelfishpg_tsql.explain_wal | off | Include information on WAL record generation | bool | off | off | FALSE |
+
+You can use the PostgreSQL [set_config()](https://www.postgresql.org/docs/14/functions-admin.html#FUNCTIONS-ADMIN-SET) function to set a parameter value on the command line. Specify the following arguments:
+
+```
+SELECT set_config(‘setting_name’, ‘value’, is_local); 
+```
+
+Where:
+
+`'setting_name'` is the name of the parameter you want to set.
+`'value'` is the parameter value.
+`is_local` : specify `true` if you would like the parameter to revert to the original setting when the transaction completes; set to `false` to make the setting persistent to the end of the session.
+
+For example, the following command sets `babelfishpg_tsql.explain_verbose` parameter to `on`; the setting reverts to the original value when the session ends:
```

Comment:
I suggest clarifying a bit more on the above two lines. "when the transactions completes" and "to the end of the session" are mentioned first. Then we get "when the session ends", which sounds like #1 but probably refers to #2. I suggest using identical wording to take out the need to interpret these words.